test/README.md

```diff
@@ -0,0 +1,46 @@
+# Tests
+
+This folder contains automated tests for this Module. All of the tests are written in [Go](https://golang.org/).
+Most of these are "integration tests" that deploy real infrastructure using Terraform and verify that infrastructure
+works as expected using a helper library called [Terratest](https://github.com/gruntwork-io/terratest).
+
+
+
+## WARNING WARNING WARNING
+
+**Note #1**: Many of these tests create real resources in an AWS account and then try to clean those resources up at the end of a test run. That means these tests may cost you money to run! When adding tests, please be considerate of the resources you create and take extra care to clean everything up when you're done!
+
+**Note #2**: Never forcefully shut the tests down (e.g. by hitting `CTRL + C`) or the cleanup tasks won't run!
```

Comment:
I did a little bit of research on this... [Terratest docs](https://terratest.gruntwork.io/docs/testing-best-practices/cleanup/) recommend using a test "nukeable" account so that [cloud-nuke](https://github.com/gruntwork-io/cloud-nuke) could run from time to time. Downside is that cloud-nuke only works with AWS. About state files, backend configs have not been set in the current code, but it is copying to a temp folder. Few options are:
- If working with those local files, we could run `terraform destroy` from it, being mindful of when/how to call (since we're dealing with containers).
- update the code to input Backend Config into the struct `terratest.Options` to use a GCS bucket to store them, and then these config values could be available from env vars or a local file. Then we'd be able to have some sort of make clean that you mentioned.

Reply:
Seems like there's a lot of solutions here. Let's put something in the backlog to get a design together to handle this case.

Reply:
Created https://datatamr.atlassian.net/browse/CA-58 to address that